app/controllers/forms/base_controller.rb

```diff
@@ -44,7 +48,17 @@ def default_url_options
     end
 
     def check_available
-      return if current_form.start_page && (mode.preview? || current_form.live?)
+      begin
+        if current_form.start_page && (mode.preview? || current_form.live?)
+          return
+        end
+      rescue ActiveResource::ResourceNotFound
+        form = archived_form
+        if form.present?
+          render template: "forms/archived/show", locals: { form_name: form.name }, status: :gone
+          return
+        end
+      end
```

Comment:
This could be made a bit less verbose and decrease indentation by using inline if statements 

```suggestion
      begin
        return if current_form.start_page && (mode.preview? || current_form.live?)
      rescue ActiveResource::ResourceNotFound
        form = archived_form
        return render template: "forms/archived/show", locals: { form_name: form.name }, status: :gone if form.present?
      end
```